src/web/components/ApiKeyManagement/keyCreationDialog.spec.tsx

```diff
@@ -0,0 +1,242 @@
+import { composeStories } from '@storybook/testing-react';
+import { fireEvent, render, screen, waitFor } from '@testing-library/react';
+import { close } from 'fs';
+
+import { ApiKeySecretsDTO } from '../../../api/services/apiKeyService';
+import KeyCreationDialog from './KeyCreationDialog';
+import * as stories from './KeyCreationDialog.stories';
+
+const { MultipleRoles, OneRole } = composeStories(stories);
+
+const writeText = jest.fn();
+
+Object.assign(navigator, {
+  clipboard: {
+    writeText,
+  },
+});
+
+describe('Key creation dialog', () => {
+  it('Should show one apiRole', () => {
+    render(<OneRole />);
+
+    const openButton = screen.getByText('Open');
+    fireEvent.click(openButton);
```

Comment:
I think we should pretty much be exclusively using `userEvent` instead of `fireEvent`, see https://testing-library.com/docs/user-event/intro/#:~:text=fireEvent%20dispatches%20DOM%20events%2C%20whereas,additional%20checks%20along%20the%20way.